project build/soong/
diff --git a/build/soong/android/rule_builder.go b/build/soong/android/rule_builder.go
index 01fe6d8ea..dd0aa9af5 100644
--- a/build/soong/android/rule_builder.go
+++ b/build/soong/android/rule_builder.go
@@ -624,6 +624,15 @@ func (r *RuleBuilder) build(name string, desc string) {
		nsjailCmd.WriteString(" -R /dev")
		nsjailCmd.WriteString(" -R /usr")

+		// PATCH: the directories above (/bin, /lib, etc) only contain symlinks to the nix store
+		// in our case. So let's mount the hosts nix store into the sandbox as well.
+		// We also saw that binaries in the sandbox didn't find libraries in /lib. This is probably
+		// due to the nixos linker (ldd) doesn't include it by default and we don't have /etc/ld.so.cache
+		// as we don't mount /etc to the sandbox. So we just set LD_LIBRARY_PATH, which seems to work.
+		nsjailCmd.WriteString(" -R /nix")
+		nsjailCmd.WriteString(" -E LD_LIBRARY_PATH=/lib")
+		// ENDPATCH
+
		nsjailCmd.WriteString(" -m none:/tmp:tmpfs:size=1073741824") // 1GB, should be enough
		nsjailCmd.WriteString(" -D nsjail_build_sandbox")
		nsjailCmd.WriteString(" --disable_rlimits")
